internal/table/table.go

```diff
@@ -55,86 +55,81 @@ func (t *Table) Render() string {
-				rowValueSize = utf8.RuneCountInString(rowValue)
-			}
-			if colWidths[n] < rowValueSize {
-				if len(row.columns) < len(t.headers) {
-					spanWidth += rowValueSize + (padding * 2) // This is a spanned column, so its width does not apply to the individual column
-				} else {
-					colWidths[n] = rowValueSize // Set width according to column size
-				}
+			columnValue, ok := sliceutils.GetString(row.columns, n)
+			if !ok {
+				continue // column doesn't exit because the previous column spans
 			}
-			if spanWidth > minWidth {
-				minWidth = spanWidth
+			columnSize := utf8.RuneCountInString(columnValue)
+
+			// Detect spanned column info
+			rowHasSpannedColumn := len(row.columns) < len(t.headers)
+			spannedColumnIndex := len(row.columns) - 1
+
+			if rowHasSpannedColumn && n == spannedColumnIndex {
+				// Record total row size as minTableWidth
+				minTableWidth = mathutils.MaxInt(minTableWidth, colWidths[n]+columnSize+(padding*2))
+			} else {
+				// This is a regular non-spanned column
+				colWidths[n] = mathutils.MaxInt(colWidths[n], columnSize)
 			}
 		}
 
 		// Add padding and update the total width so far
 		colWidths[n] += padding * 2
-		columnTotal += colWidths[n]
+		colWidthsCombined += colWidths[n]
 	}
 
 	// Equalize widths by 20% of average width
 	// This is to prevent columns that are much larger than others from taking up most of the table width
-	colWidths, columnTotal = equalizeWidths(colWidths, columnTotal, 20)
+	equalizeWidths(colWidths, 20)
 
-	// compute the total number of columns that we want the table to use
-	targetTotal := columnTotal
-	// Factor in spanned columns
-	if targetTotal < minWidth {
-		targetTotal = minWidth
-	}
+	// Constrain table to max and min dimensions
+	tableWidth := mathutils.MaxInt(colWidthsCombined, minTableWidth)
+	tableWidth = mathutils.MinInt(tableWidth, maxTableWidth)
+
+	// Now scale back the row sizes according to the max width
+	rescaleColumns(colWidths, tableWidth)
+	logging.Debug("Table column widths: %v, total: %d", colWidths, tableWidth)
+
+	return colWidths, tableWidth
+}
+
+// equalizeWidths equalizes the width of given columns by a given percentage of the average columns width
+func equalizeWidths(colWidths []int, percentage int) {
+	total := float64(mathutils.Total(colWidths...))
+	multiplier := float64(percentage) / 100
+	averageWidth := total / float64(len(colWidths))
 
-	// Limit to max width
-	if targetTotal > maxTotalWidth {
-		targetTotal = maxTotalWidth
+	for n := range colWidths {
+		colWidth := float64(colWidths[n])
+		colWidths[n] += int((averageWidth - colWidth) * multiplier)
 	}
 
-	colWidths = rescaleColumns(colWidths, columnTotal, targetTotal)
-	logging.Debug("Table column widths: %v, total: %d", colWidths, targetTotal)
+	// Account for floats that got rounded
+	colWidths[len(colWidths)-1] += int(total) - mathutils.Total(colWidths...)
+}
+
+func rescaleColumns(colWidths []int, targetTotal int) {
```

Comment:
This is okay, and certainly simpler to read.  I rewrote in a way such that never more than one extra column would be added to the last column.  If we have six columns and the first five columns were all truncated down by .8, the last column what get an extra four characters (4 = 5 * 0.8).  That seems excessive. :) But of course, it is also a total edge case, and we will rarely have to work with small column widths where this makes a difference.

Reply:
Yeah I'd favour readability over accounting for a few units that don't make much of a difference in the grand scheme of things.